src/components/TableOfContents/index.tsx

```diff
@@ -1,83 +1,64 @@
 'use client'
 
-import type { ListItem } from '@components/Jumplist/types'
 import type { Heading } from '@root/collections/Docs/types'
 
-import React, { useEffect, useRef, useState } from 'react'
+import React, { useEffect, useReducer } from 'react'
 
-import { Jumplist } from '../Jumplist/index'
 import classes from './index.module.scss'
 
 export type Props = {
   className?: string
   headings: Heading[]
 }
 
-export const TableOfContents: React.FC<Props> = ({ className, headings }) => {
-  const listItemRefs = useRef<(HTMLDivElement | null)[]>([])
-  const [indicatorTop, setIndicatorTop] = useState<number | undefined>(undefined)
-  const [activeHeadingId, setActiveHeadingId] = useState<null | string>(null)
-  const [resetIndicator, setResetIndicator] = useState(true)
+/**
+ * returns the height at which the ToC indicator should be.
+ * Currently the top position. We could make it cover all visible headings, as fumadocs does.
+ */
```

Comment:
A "nice to have" feature for another day. I don't think it's that much of a priority right now.